core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -134,15 +140,15 @@ protected int run() throws Exception {
             a = new ParametersAction(values);
         }
 
-        if (checkSCM) {
-            if (job.poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
+        if (checkSCM && job instanceof AbstractProject<?, ?>) {
+            if (((AbstractProject<?, ?>)job).poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
```

Comment:
You can run polling in general by using `SCMTriggerItems.asSCMTriggerItem(job).poll(listener).hasChanges()`